swarm.go

```diff
@@ -374,11 +374,24 @@ func (s *Swarm) Start(srv *p2p.Server) error {
 	log.Info("Updated bzz local addr", "oaddr", fmt.Sprintf("%x", newaddr.OAddr), "uaddr", fmt.Sprintf("%s", newaddr.UAddr))
 
 	if s.config.SwapEnabled {
-		err := s.swap.Deploy(context.Background(), s.backend, s.config.Path)
-		if err != nil {
-			return fmt.Errorf("Unable to deploy swap contract: %v", err)
+		if s.config.Contract != (common.Address{}) {
+			address := s.config.Contract
+			err := cswap.ValidateCode(context.Background(), s.backend, address)
+			if err != nil {
+				return fmt.Errorf("Provided address not a chequebook smart-contract: %v", err)
```

Comment:
`err` could be returned due to two things:
* `CodeAt` failed
* `ErrNotASwapContract`

For the second case, the error formatting would not add information and would be redundant, for the first case it would be slightly misleading. I would change this to be 
`return fmt.Errorf("contract validation for %v failed: %v", address, err)`
or even just returning `return err`

Reply:
can i just express how scary it is that we made almost _the exact same_ comment? :fearful: